radixdlt/babylon-wallet-android
# [ABW-2263] ]Crash reporting setup

## Description
Crash reporting setup is available only in `debugAlpha` and `release` build types. By default user is opted-out, it can be enabled in App Settings.
Notes:
- to build that branch you need to download up to date `google-services.json` from wallet firebase console and place it in app folder
- I've added logNonFatal method and called in in few places in dapp login and transaction submit flows where we catch and handle exceptions. I verified that those non fatal exceptions are not logged when crash reporting is disabled, but we can also think about wrapping that method call into a use case which will also check local data store flag `KEY_CRASH_REPORTING_ENABLED`


## How to test

To test it you would need to build debugAlpha/release or switch build.gradle crash reporting flag for debug build to `true`, place code that will crash the app somewhere and try that with crash reporting enabled/disabled, to verify if crashes are sent (or not sent) to specific project in firebase console. Propagation time that I observed was fast, less then a minute.

## PR submission checklist
- [ ] I have tested that crash reporting is only available in specific builds, and that no crash reports are being sent when reporting is disabled
